src/content/learn/choosing-the-state-structure.md

```diff
@@ -1,53 +1,53 @@
-2. **Avoid contradictions in state.** When the state is structured in a way that several pieces of state may contradict and "disagree" with each other, you leave room for mistakes. Try to avoid this.
-3. **Avoid redundant state.** If you can calculate some information from the component's props or its existing state variables during rendering, you should not put that information into that component's state.
-4. **Avoid duplication in state.** When the same data is duplicated between multiple state variables, or within nested objects, it is difficult to keep them in sync. Reduce duplication when you can.
-5. **Avoid deeply nested state.** Deeply hierarchical state is not very convenient to update. When possible, prefer to structure state in a flat way.
+1. **Bağlantılı state değişkenlerini gruplayın.** Eğer birden fazla state değişkenini hep aynı anda güncelliyorsanız, bu değişkenleri tek bir state değişkeninde birleştirmeyi değerlendirin.
+2. **State çelişkilerinden kaçının.** Eğer bir bileşenin state değişkenleri birbiriyle çelişecek ya da uyuşmayacak şekilde tasarlanırsa hatalara açık kapı bırakmış olursunuz. Bu durumdan kaçının.
+3. **Gereksiz state oluşturmaktan kaçının.** Eğer bileşeni render ederken, bileşenin prop'larından ya da varolan state değişkenlerinden ihtiyacınız olan bilgiyi hesaplayabiliyorsanız bu bilgiyi bileşenin state'inde tutmamalısınız.
+4. **Yinelenen state değişkenlerinden kaçının.** Aynı veri, farklı state değişkenleri ya da iç içe nesneler içinde yinelendiğinde senkronizasyonu sağlamak zorlaşabilir. Buna benzer tekrarları en aza indirgemeye çalışın.
+5. **Derinlemesine iç içe olan bir state yapısından kaçının.** Derinlemesine hiyerarşik yapıya sahip olan state'i güncellemek çok da pratik bir değildir. State'i daha düz bir yapıda şekillendirmeye çalışın.
 
-The goal behind these principles is to *make state easy to update without introducing mistakes*. Removing redundant and duplicate data from state helps ensure that all its pieces stay in sync. This is similar to how a database engineer might want to ["normalize" the database structure](https://docs.microsoft.com/en-us/office/troubleshoot/access/database-normalization-description) to reduce the chance of bugs. To paraphrase Albert Einstein, **"Make your state as simple as it can be--but no simpler."**
+Bu prensiplerin asıl amacı, *state'i hatalara sebep olmadan kolayca güncelleyebilmektir*. State'ten gereksiz ve yinelenen veriyi temizlemek tüm parçaların birbiriyle senkronize kalmasına yardımcı olur. Bu durum, bir veritabanı mühendisinin hata ihtimalini azaltmak için [veritabanı yapısını "normalleştirmek"](https://learn.microsoft.com/tr-tr/office/troubleshoot/access/database-normalization-description) istemesine benzetilebilir. Albert Einstein'ın sözleriyle yorumlamak gerekirse, **"State mümkün olabildiğince basit olmalı, sadece basit değil."**
```

Comment:
Bunun orijinali `Make everything as simple as possible, but not simpler.` 

Yani aslinda, `mevzulari anlasilabilecek kadar basitlestirin ama cok fazla basit hale getirmeyin. O zaman da anlamini kaybeder, sacma birsey olur. ` demek istiyor. O yuzden  son kismi daha iyi cevrilebilir:

`State'i olabildiğince basit hale getirin; ama çok da basit olmasın.`

Tarzi bisey olabilir.

Reply:
Evet bu şekilde daha anlaşılır oluyor 👍